src/xunit.performance.analysis/Program.cs

```diff
@@ -43,256 +41,179 @@ private static int Main(string[] args)
 
+                        if(baselineCount <= 0 || comparisonCount <= 0)
+                        {
+                            continue;
+                        }
+                        
+                        if(baselineTest  .Iterations[0].MetricValues.ContainsKey( metricName ) == false || 
+                           comparisonTest.Iterations[0].MetricValues.ContainsKey( metricName ) == false  )
+                        {
+                            continue;
+                        }
 
-        private static void WriteStatisticsCSV(Dictionary<string, Dictionary<string, TestResult>> testResults, string analyzeOutputPath)
-        {
-            using (var writer = new StreamWriter(analyzeOutputPath))
-            {
-                writer.WriteLine("Test, Iterations, Duration Min, Duration Max, Duration Average, Duration Stdev, Metrics");
-                foreach (var run in testResults)
-                {
-                    foreach (var result in run.Value.Values)
-                    {
-                        RunningStatistics durationStats = result.Stats[DurationMetricName].RunningStatistics;
-                        writer.WriteLine(
-                            "\"{0}\",\"{1}\",\"{2}\",\"{3}\",\"{4}\",\"{5}\",\"{6}\"",
-                            EscapeCsvString(result.TestName),
-                            durationStats.Count,
-                            durationStats.Minimum,
-                            durationStats.Maximum,
-                            durationStats.Mean,
-                            durationStats.StandardDeviation,
-                            EscapeCsvString(GetMetricsString(result.Stats.Keys))
-                            );
+                        // Compute the standard error in the difference
+                        var baselineSum = baselineTest.Iterations.Sum(iteration => iteration.MetricValues[metricName]);
+                        var baselineSumSquared = baselineSum * baselineSum;
+                        var baselineSumOfSquares = baselineTest.Iterations.Sum(iteration => iteration.MetricValues[metricName] * iteration.MetricValues[metricName]);
+                        
+                        var comparisonSum = comparisonTest.Iterations.Sum(iteration => iteration.MetricValues[metricName]);
+                        var comparisonSumSquared = comparisonSum * comparisonSum;
+                        var comparisonSumOfSquares = comparisonTest.Iterations.Sum(iteration => iteration.MetricValues[metricName] * iteration.MetricValues[metricName]);
+
+                        var stdErrorDiff = Math.Sqrt((baselineSumOfSquares - (baselineSumSquared / baselineCount) + comparisonSumOfSquares - (comparisonSumSquared / comparisonCount)) * (1.0 / baselineCount + 1.0 / comparisonCount) / (baselineCount + comparisonCount - 1));
+                        var interval = stdErrorDiff * MathNet.Numerics.ExcelFunctions.TInv(1.0 - Properties.ErrorConfidence, baselineCount + comparisonCount - 2);
```

Comment:
We're assuming here that the data here has a normal distribution. This is unlikely for performance data, especially given the generally small default sample size (10).

I realize you're just reworking preexisting code, but we should consider using a non-parametric estimation technique for metrics on populations, and some variant of hypothesis-based testing for determining if some metric is significantly different between populations.